source/dhtslib/vcf/record.d

```diff
@@ -14,18 +14,278 @@ import core.vararg;
 
 import std.conv: to, ConvException;
 import std.format: format;
-import std.range: ElementType;
+import std.range: ElementType, chunks;
 import std.string: fromStringz, toStringz;
+import std.utf : toUTFz;
+import std.algorithm : map;
+import std.array : array;
 import std.traits;
+import std.meta : staticIndexOf;
 
+import dhtslib.vcf;
 import dhtslib.coordinates;
-import dhtslib.vcf.header;
 import htslib.hts_log;
 import htslib.kstring;
 import htslib.vcf;
 
 alias BCFRecord = VCFRecord;
 
+/// Struct to aid in conversion of VCF info data
+/// into D types
+struct InfoField
```

Comment:
So the reason for copying is because of data ownership. And yes it can be done without a copy, the copy is an easy fix for some edge cases where you reference the info data but the `VCFRecord` is allowed to go out of scope or the `dtor` allowed to be called. In that situation, the info data is obviously gone and any references are invalid. An alternate to this implementation is similar to what I have done with `SAMRecord`, `Cigar`, and `TagValue`. All structs get reference counting, `VCFRecord` would have to check the reference counts of the `InfoFields` and `FormatFields` at destruction to make sure they didn't have references before the `bcf1_t` is destroyed in the `dtor`. I can certainly implement this.

Reply:
I have no objection to VCFHeader, and VCFHeader records being copied, but for the VCFRecord itself (and INFO and FORMAT) there will be a huge perofrmance hit when say parsing Gnomad etc. Probably this s hould mirror SAMRecord

Reply:
addressed in
e91d936